CASMINST-7373: adding job yaml and script for k8s upgrade (#6141)
Cray-HPE/docs-csm: operations/iuf/workflows/deploy_product.md

Once this step has completed:

> **NOTE**  

> > In CSM 1.7, as part of the `deploy-product` stage during CSM upgrade:  

Reviewer: ```suggestion
> As part of the `deploy-product` stage during upgrades from CSM 1.6 to CSM 1.7:  
```

>
> - Kubernetes will be upgraded from version 1.26 to 1.32.5.  
> - The `deploy-product-onexit` hook will launch a Kubernetes upgrade job that runs outside of IUF, in the `argo` namespace.  
> - This job must be monitored manually and must complete successfully before proceeding to the next stage.  
>
> The output from the `deploy-product` stage will look like:
>

> > ```bash

Reviewer: ```suggestion
> ```text
```

> INFO Job upgrade-k8s-job-zm55x has been created in the argo namespace. This is performing k8s upgrade from 1.26 to 1.32
> INFO Monitor the job and ensure it is successful before proceeding to next stage.
> ```  
>

> > To monitor the job, run:

Reviewer: Whenever a command is specified, the preceding text needs to include this markup to specify where it should be run.
See [Command prompt conventions](https://github.com/Cray-HPE/docs-csm/blob/release/1.7/introduction/documentation_conventions.md#command-prompt-conventions) for more information.
```suggestion
> (`ncn-mw#`) To monitor the job, run:
```

Author: done
